### ISSUE TITLE
print function does not work within rust

### ISSUE THREAD

**Issue by lorencarvalho** [PyO3/pyo3#660]
### 🌍 Environment

 - Your operating system and version: macOS Mojave 10.14.6 (18G95)
 - Your python version: 3.7.4
 - How did you install python (e.g. apt or pyenv)? macOS installer via python.org
- Did you use a virtualenv?: No, n/a as I'm calling Python from Rust.
 - Your rust version (`rustc --version`): rustc 1.39.0-nightly (17e73e801 2019-08-28)
 - Are you using the latest pyo3 version? Yes
- Have you tried using latest master (replace `version = "0.x.y"` with `git = "https://github.com/PyO3/pyo3")?` yes

### 💥 Reproducing

```
use pyo3::prelude::*;
use pyo3::types::PyDict;

fn main() {
    let gil = Python::acquire_gil();
    let py = gil.python();
    let locals = PyDict::new(py);
    py.run("foo = 'one'; print(foo)", None, Some(locals));
    py.run("print(\"hello python\"", None, None);
    println!("hello {}", locals.get_item("foo").unwrap());
}
```

Hello! It seems that calling `print()` in a Python snippet from within Rust does not work. I thought that perhaps Pyo3 did something with stdout/stderr file descriptors, but calling `import code; code.interact()` via a `py_run!` macro (surprisingly) does work, and everything seems mostly normal (save for sys.argv being an empty list and a few other expected oddities). 

Obviously this isn't super high priority, but I figured I'd mention it :)

Thanks for such a cool project! Really excited to work with it.

**Comment by programmerjake** [comment]
> ```
>     py.run("print(\"hello python\"", None, None);
> ```

you're missing the closing parenthesis `)` in the print function call.

didn't test if that fixes it.

**Comment by lorencarvalho** [comment]
@programmerjake it doesn't, I just bungled copying my example code

**Comment by ehiggs** [comment]
It works here.
Ubuntu 19.10, python3 installed using apt. No venv. Rust 1.40 nightly 2019-11-7. pyo3 0.8.2.

```
$ python3 --version
Python 3.7.5rc1
```

```rust
$ cat src/main.rs 
use pyo3::prelude::*;
use pyo3::types::PyDict;

fn main() {
    let gil = Python::acquire_gil();
    let py = gil.python();
    let locals = PyDict::new(py);
    py.run("foo = 'one'; print(foo)", None, Some(locals)).expect("smash");
    py.run("print(\"hello python\")", None, None).expect("bang");
    println!("hello {}", locals.get_item("foo").unwrap());
}
```
```
$ ./target/debug/pyo3-test 
one
hello python
hello one
```

**Comment by programmerjake** [comment]
maybe it would work by using `print(..., flush=True)`?

**Comment by davidhewitt** [comment]
I think this was probably resolved by #943, so I'm going to close. If anyone encounters this and can provide another repro let's reopen.
